src/Psalm/Internal/Analyzer/Statements/Expression/CastAnalyzer.php

```diff
@@ -59,37 +58,33 @@ public static function analyze(
                     if (!$maybe_type->from_calculation) {
                         self::handleRedundantCast($maybe_type, $statements_analyzer, $stmt);
                     }
+
+                    if ($maybe_type->isSingleStringLiteral()) {
+                        $valid_int_type = new Type\Union([
+                            new Type\Atomic\TLiteralInt((int)$maybe_type->getSingleStringLiteral()->value),
```

Comment:
you can use `Type;;getInt(false, (int)$maybe_type->getSingleStringLiteral()->value);`